submariner/IPV6-datapath.md

```diff
@@ -0,0 +1,129 @@
+# Submariner Enhancement for IPV6 datapath
+
+<!-- Add link to issue/epic if available -->
+
+## Summary
+
+IPv4, IPV6 and dual-stack networking is supported for Kubernetes cluster starting in 1.21.
+IPV6 networking allowing assignment of IPv6 addresses.
+Dual-stack networking allowing the simultaneous assignment of both IPv4 and IPv6 addresses.
+
+IPv4/IPv6 dual-stack on your Kubernetes cluster provides the following features:
+
+* Dual-stack Pod networking (a single IPv4 and IPv6 address assignment per Pod)  
+* IPv4 and IPv6 enabled Services  
+* Pod off-cluster egress routing (eg. the Internet) via both IPv4 and IPv6 interfaces
+
+Currently, Submariner only supports IPV4 datapath and this proposal explains the changes required to support IPV6 and dual-stack clusters.
+
+## Inter-cluster datapath requirements
+
+The next table describes the required inter-cluster connectivity for the various combinations of clusters networking configurations:  
```

Comment:
What if we have more than 2 clusters - f.e., lets say we have three clusters A, B, C
A is dualStack
B is IPv4 only
C is IPv6 only

Even if A connects to B via IPv4 and to C via IPv6, the connectivity between B and C will not work. Taking this as an example, can you please include notes on how Lighthouse would support service discovery in a hybrid environment - as advertising v6 addresses to a v4 cluster (vice versa) will not help/work.

Reply:
I would like you to consider another use-case. Lets say we have just two clusters D and E.
D is a dualStack cluster
E is an IPv4 only cluster.

In this scenario, if we establish IPv4 tunnels between the cluster, and if the user exports an IPv6 only service (by setting .spec.ipFamilies to "IPv6"), how will Submariner handle this scenario? Does it ignore such exported services from advertising to the Broker or is it expected that cluster E would get the serviceImport details from Broker and will ignore it as it has only IPv6 address and not v4.